flang/lib/Optimizer/Transforms/AddDebugInfo.cpp

```diff
@@ -139,6 +146,8 @@ mlir::LLVM::DIModuleAttr AddDebugInfoPass::getOrCreateModuleAttr(
 void AddDebugInfoPass::handleGlobalOp(fir::GlobalOp globalOp,
                                       mlir::LLVM::DIFileAttr fileAttr,
                                       mlir::LLVM::DIScopeAttr scope) {
+  if (mlir::isa<mlir::FusedLoc>(globalOp.getLoc()))
+    return;
```

Comment:
Could you share that into a `debugInfoIsAlreadySet(globalOp.getLoc())` static helper so that there it is updated too when https://github.com/llvm/llvm-project/pull/95862 lands.

Reply:
Done.